lint/comments_checker.go

```diff
@@ -0,0 +1,33 @@
+package lint
+
+import (
+	"go/ast"
+	"regexp"
+)
+
+type commentsChecker struct {
+	ctx    *context
+	regexp *regexp.Regexp
+}
+
+func commentsCheck(ctx *context) func(*ast.File) {
+	re := `//\s?\w+[^a-zA-Z]+$`
+	c := commentsChecker{ctx: ctx, regexp: regexp.MustCompile(re)}
+	return c.CheckFile
+}
+
+func (c *commentsChecker) CheckFile(f *ast.File) {
+	for _, decl := range f.Decls {
+		if decl, ok := decl.(*ast.FuncDecl); ok {
+			if decl.Doc != nil {
+				if c.regexp.MatchString(decl.Doc.List[0].Text) {
+					c.warn(decl)
+				}
+			}
```

Comment:
You can do something like this:
```go
- if decl.Doc != nil {
- 	if c.regexp.MatchString(decl.Doc.List[0].Text) {
- 		c.warn(decl)
- 	}
- }
+ if decl.Doc != nil && c.regexp.MatchString(decl.Doc.List[0].Text) {
+ 	c.warn(decl)
+ }
```
But this is a matter of taste and preference.

We can ask what @fexolm prefers here. :)

Reply:
I prefer second one if line wouldn't be so long.
If so, you could use smth like
```
cond := decl.Doc != nil &&
        c.regexp.MatchString(...)
if cond {
...
}
```